vaadin-date-picker-flow/src/main/resources/META-INF/resources/frontend/datepickerConnector.js

```diff
@@ -78,69 +132,176 @@
                     console.warn("The locale is not supported, using default locale setting(en-US).");
                 }
 
-                let currentDate = false;
+                /** @type {DateHash} */
+                let currentDate;
+                /** @type {string} */
                 let inputValue = getInputValue();
-                if (datepicker.i18n.parseDate !== 'undefined' && inputValue) {
+
+                if (datepicker.i18n.parseDate && inputValue) {
```

Comment:
There was a bug here (noticed by the IDE after adding type info). `datepicker.i18n.parseDate !== 'undefined'` was always true since `parseDate` is never set to the string `'undefined'`.

Could have used `parseDate !== undefined` but I think this new version is nicer in this case (and shorter)